### PR TITLE
Update openapi-gen to not rely on vendor

### PR DESCRIPTION
**Pull request by Jeffwan**
1. Update `gen-sdk.sh` to use kube-openapi from go modules rather than vendor. Not all `code-generator` has `/cmd/openapi-gen`, 1.15 doesn't while it's added back in 1.16. Let's use kube-openapi here. 
2. `central.maven.org` is not resolvable on my end, seems it's http legacy url. Update the jar url and points to `repo1.maven.org`

### CODE REVIEW

File: hack/python-sdk/gen-sdk.sh [hunk 1]
```diff
@@ -28,8 +28,22 @@ if [ -z "${GOPATH:-}" ]; then
...
+    go mod download
+fi
+
+echo ">> Using ${OPENAPI_PKG}"
+
+echo "Building openapi-gen"
+go build -o openapi-gen ${OPENAPI_PKG}/cmd/openapi-gen
+
 echo "Generating OpenAPI specification ..."
-go run vendor/k8s.io/code-generator/cmd/openapi-gen/main.go --input-dirs github.com/kubeflow/pytorch-operator/pkg/apis/pytorch/v1,github.com/kubeflow/common/job_controller/api/v1 --output-package github.com/kubeflow/pytorch-operator/pkg/apis/pytorch/v1 --go-header-file hack/boilerplate/boilerplate.go.txt
```
Comment (Jeffwan) [inline hack/python-sdk/gen-sdk.sh:32]: @jinchihe  Do we need to add`github.com/kubeflow/common/job_controller/api/v1` as input? I actually have some failure if I add common API `github.com/kubeflow/common/job_controller/api/v1` like below. Have you ever seen the problem? 

```
$ ./openapi-gen --input-dirs github.com/kubeflow/common/job_controller/api/v1,github.com/kubeflow/pytorch-operator/pkg/apis/pytorch/v1 --output-package github.com/kubeflow/pytorch-operator/pkg/apis/pytorch/v1 --go-header-file hack/boilerplate/boilerplate.go.txt

E0519 14:11:58.053066   14182 openapi.go:294] Error when generating: PodInformerSynced, PodInformerSynced k8s.io/client-go/tools/cache.InformerSynced
2020/05/19 14:11:58 OpenAPI code generation error: Failed executing generator: some packages had errors:
cannot generate spec for type k8s.io/client-go/tools/cache.InformerSynced

```

Comment (gaocegege) [reply hack/python-sdk/gen-sdk.sh:32]: cc @jinchihe

Comment (jinchihe) [reply hack/python-sdk/gen-sdk.sh:32]: Sorry I missed the PR since I was on vacation last month, will take a look asap. Thanks,

Comment (jinchihe) [reply hack/python-sdk/gen-sdk.sh:32]: > @jinchihe Do we need to add`github.com/kubeflow/common/job_controller/api/v1` as input? I actually have some failure if I add common API `github.com/kubeflow/common/job_controller/api/v1` like below. Have you ever seen the problem?
> 
> ```
> $ ./openapi-gen --input-dirs github.com/kubeflow/common/job_controller/api/v1,github.com/kubeflow/pytorch-operator/pkg/apis/pytorch/v1 --output-package github.com/kubeflow/pytorch-operator/pkg/apis/pytorch/v1 --go-header-file hack/boilerplate/boilerplate.go.txt
> 
> E0519 14:11:58.053066   14182 openapi.go:294] Error when generating: PodInformerSynced, PodInformerSynced k8s.io/client-go/tools/cache.InformerSynced
> 2020/05/19 14:11:58 OpenAPI code generation error: Failed executing generator: some packages had errors:
> cannot generate spec for type k8s.io/client-go/tools/cache.InformerSynced
> ```

@Jeffwan I do not remember hit the problem. If the problem can be resolved by adding the input , yeah, please go ahead.

File: hack/python-sdk/gen-sdk.sh [hunk 2]
```diff
@@ -18,7 +18,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-SWAGGER_JAR_URL="http://central.maven.org/maven2/io/swagger/swagger-codegen-cli/2.4.6/swagger-codegen-cli-2.4.6.jar"
+SWAGGER_JAR_URL="https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/2.4.6/swagger-codegen-cli-2.4.6.jar"
```
Comment (jinchihe) [inline hack/python-sdk/gen-sdk.sh:21]: Would you mind to rebase to remove this change. I did not realized the PR, and already fixed this last week. Thanks a lot!

Comment (Jeffwan) [reply hack/python-sdk/gen-sdk.sh:21]: Sure. I will do.

**Comment by Jeffwan** [comment]
@jinchihe Please have another check

**Comment by jinchihe** [comment]
/lgtm 

@Jeffwan Thanks!

**Comment by johnugeorge** [comment]
Sorry for being late
/approve

**Comment by Jeffwan** [comment]
Test failure is probably related with disk out of capacity failure.

**Pull request merged**